src/core/thread.d

```diff
@@ -1709,13 +1709,22 @@ private:
         return cast(Mutex)_locks[1].ptr;
     }
 
-    __gshared align(Mutex.alignof) void[__traits(classInstanceSize, Mutex)][2] _locks;
+    // Handling unaligned mutexes are not supported on all platforms, so we
+    // must round up the statically allocated size of the _locks array to
+    // ensure that the address of all elements are appropriately aligned.
+    import core.internal.traits : classInstanceAlignment;
+
+    enum mutexAlign = classInstanceAlignment!Mutex;
+    enum alignedMutexInstanceSize = (__traits(classInstanceSize, Mutex) + mutexAlign - 1) & ~(mutexAlign - 1);
+
+    __gshared align(mutexAlign) void[alignedMutexInstanceSize][2] _locks;
```

Comment:
This can be simplified like this:
```D
struct RawStorage(size_t size, ubyte alignment)
{ align (alignment) void[size] storage; }

alias MutexStorage = RawStorage!(
    __traits(classInstanceSize, Mutex),
    classInstanceAlignment!Mutex
);

__gshared MutexStorage[2] _locks;
```
...

```D
static void initLocks() @nogc
{
    foreach (ref lock; _locks)
    {
        lock.storage[] = typeid(Mutex).initializer[];
        (cast(Mutex)lock.storage.ptr).__ctor();
    }
}
```